Stratis.Bitcoin/Logging/LogsExtension.cs

```diff
@@ -7,99 +7,181 @@
+
+            // If we use debugFile target, which is defined in "NLog.config", we make sure it is in the log folder.
+            Target debugTarget = LogManager.Configuration.FindTargetByName("debugFile");
+            if (debugTarget != null)
+            {
+                // Extracts the name of the file currently used and puts it into the log folder.
+                string currentFile = (debugTarget as FileTarget).FileName.Render(new LogEventInfo { TimeStamp = DateTime.UtcNow });
+                (debugTarget as FileTarget).FileName = Path.Combine(folder.LogPath, Path.GetFileName(currentFile));
+            }
+
+            // Remove rule that forbids logging before the logging is initialized.
+            LoggingRule nullPreInitRule = null;
+            foreach (LoggingRule rule in LogManager.Configuration.LoggingRules)
+            {
+                if (rule.Final && rule.NameMatches($"*") && (rule.Targets.Count > 0) && (rule.Targets[0].Name == "null"))
+                {
+                    nullPreInitRule = rule;
+                    break;
+                }
+            }
+            LogManager.Configuration.LoggingRules.Remove(nullPreInitRule);
 
+
+            // Configure main file target, configured using command line or node configuration file settings.
+            var mainTarget = new FileTarget();
+            mainTarget.Name = "main";
+            mainTarget.FileName = Path.Combine(folder.LogPath, "node.txt");
+            mainTarget.ArchiveFileName = Path.Combine(folder.LogPath, "node-${date:universalTime=true:format=yyyy-MM-dd}.txt");
+            mainTarget.ArchiveNumbering = ArchiveNumberingMode.Sequence;
+            mainTarget.ArchiveEvery = FileArchivePeriod.Day;
+            mainTarget.MaxArchiveFiles = 7;
+            mainTarget.Layout = "[${longdate:universalTime=true} ${threadid}${mdlc:item=id}] ${level:uppercase=true}: ${callsite} ${message}";
+            mainTarget.Encoding = Encoding.UTF8;
+
+            LogManager.Configuration.AddTarget(mainTarget);
+
+            // Default logging level is Info for all components.
+            var defaultRule = new LoggingRule($"{nameof(Stratis)}.{nameof(Stratis.Bitcoin)}.*", NLog.LogLevel.Info, mainTarget);
+
+            // Configure main file target rules based on node settings.
+            // TODO: Preload enough args for -conf= or -datadir= to get debug args from there. We currently forbid logging before the logging is initialized.
+            // TODO: Currently only takes -debug arg.
             var keyToCategory = new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase)
             {
                 //{ "addrman", "" },
                 //{ "alert", "" },
-                { "bench", "Stratis.Bitcoin.FullNode.ConsensusStats" },
+                { "bench", $"{nameof(Stratis)}.{nameof(Stratis.Bitcoin)}.{nameof(Stratis.Bitcoin.Features)}.{nameof(Stratis.Bitcoin.Features.Consensus)}.{nameof(Stratis.Bitcoin.Features.Consensus.ConsensusStats)}" },
```

Comment:
it works only for final classes - but except for this ConsensusStats and FullNode, everything else is just namespaces, so yes, we can have this for those two, but that would be inconsistent with others ...